File: primesearcher.py

import math
import time

def isPrime(n) :

	s = math.sqrt(n)

	i = 3
	while i<=s :
		if n % i == 0 :
			return 0
		i += 2
	return 1



candidate = 10000001

numConsidered = 0

numPrimes = 0

startTime = time.time()

while numConsidered < 5000000 :

	if isPrime(candidate) :
		numPrimes += 1

	candidate += 2
	numConsidered += 1

	if numConsidered % 1000000 == 0 :
		t = time.time() - startTime
		print t, numConsidered, numPrimes, numPrimes*100.0/numConsidered, numConsidered/t, numPrimes/t

File: primesearcher.c

#include <stdio.h>
#include <math.h>
#include <time.h>
#include <unistd.h>

int isPrime(long n) {

	long i;
	long s = (long)sqrt((double)n);

	for(i=3; i<=s; i+=2) {
		if(n % i == 0) 
			return 0;
	}

	return 1;
}

int main(int argc, char **argv) {

	long candidate = 10000001;

	long numConsidered = 0;
	long numPrimes = 0;

	clock_t startTime = clock();

	while(numConsidered < 5000000) {
		if(isPrime(candidate)) {
			numPrimes++;
		}
		candidate += 2;
		numConsidered++;

		if(numConsidered % 1000000 == 0) {
			long t = (clock() - startTime)*100/CLOCKS_PER_SEC;
			printf("%f %ld %ld %f %f %f\n",  t/100.0, numConsidered, numPrimes, 
					(double)numPrimes*100.0/(double)numConsidered, 
					(double)numConsidered/(t/100.0), (double)numPrimes/(t/100.0));

		}
	}

	return 0;
}


